### PR TITLE
segregate http client

### PR DESCRIPTION
**Pull request by bhakiyakalimuthu**
Create http client connection only once, Same tcp connection can be reused by every proxy request.Clients and Transports are safe for concurrent use by multiple goroutines and for efficiency should only be created once and re-used.
https://pkg.go.dev/net/http

- [x] Move http client to seperate file
- [x] Remove unused utility functions from utils
- [x] Move test utility methods to test folder

### CODE REVIEW

File: server/http_client.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+	"strconv"
+	"time"
+)
+
+type HttpClient interface {
+	ProxyRequest(body []byte) (*http.Response, error)
+}
+
+type httpClient struct {
+	httpClient *http.Client // http client for making proxy request
```
Comment (metachris) [inline server/http_client.go:15]: doesn't need to be a pointer

File: server/http_client.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+	proxyURL   string       // Proxies the incoming request to the target URL
```
Comment (metachris) [inline server/http_client.go:16]: doc comment is incorrect

File: server/request_processor.go [hunk 3]
```diff
@@ -315,8 +313,19 @@ func (r *RpcRequest) handleCancelTx() (requestCompleted bool) {
 func (r *RpcRequest) GetAddressNonceRange(address string) (minNonce, maxNonce uint64, err error) {
 	// Get minimum nonce by asking the eth node for the current transaction count
 	_req := types.NewJsonRpcRequest(1, "eth_getTransactionCount", []interface{}{r.txFrom, "latest"})
-	_res, err := utils.SendRpcAndParseResponseTo(r.defaultProxyUrl, _req)
+	jsonData, err := json.Marshal(_req)
 	if err != nil {
+		r.logger.Error("[sendTxToRelay] eth_getTransactionCount marshal failed", "error", err)
```
Comment (metachris) [inline server/request_processor.go:318]: this is not `sendTxToRelay`

File: server/util.go [hunk 4]
```diff
@@ -103,3 +85,59 @@ func GetTxStatus(txHash string) (*types.PrivateTxApiResponse, error) {
...
+func IsMetamask(r *http.Request) bool {
+	return r.Header.Get("Origin") == "chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn"
+}
+
+// FIREFOX_ID: webextension@metamask.io
+func IsMetamaskMoz(r *http.Request) bool {
+	return r.Header.Get("Origin") == "moz-extension://57f9aaf6-270a-154f-9a8a-632d0db4128c"
+}
+
+func ParseResponseTo(resp *http.Response) (*types.JsonRpcResponse, error) {
```
Comment (metachris) [inline server/util.go:107]: Should be called `ParseJsonRPCResponse`

File: server/http_client.go [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+func NewHttpClient(proxyURL string) HttpClient {
+	return &httpClient{
+		httpClient: &http.Client{Timeout: time.Second * 5}, // default timeout set to 5s
+		proxyURL:   proxyURL,
+	}
+}
+
+// ProxyRequest using http client to make http post request
+func (n *httpClient) ProxyRequest(body []byte) (*http.Response, error) {
+
```
Comment (metachris) [inline server/http_client.go:28]: remove unnecessary newline

File: server/http_client.go [hunk 6]
```diff
@@ -0,0 +1,40 @@
+package server
+
+import (
+	"bytes"
+	"net/http"
+	"strconv"
+	"time"
+)
+
+type HttpClient interface {
```
Comment (metachris) [inline server/http_client.go:10]: perhaps `RPCProxyClient` would be better?

File: server/http_client.go [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+type httpClient struct {
+	httpClient *http.Client // http client for making proxy request
+	proxyURL   string       // Proxies the incoming request to the target URL
+}
+
+func NewHttpClient(proxyURL string) HttpClient {
+	return &httpClient{
+		httpClient: &http.Client{Timeout: time.Second * 5}, // default timeout set to 5s
```
Comment (metachris) [inline server/http_client.go:21]: comment is unnecessary

**Comment by metachris** [comment]
PR looks good overall 👍 . Some cleanup is missing, left a couple comments.

File: server/http_client.go (hunk 1 above)
Comment (bhakiyakalimuthu) [reply server/http_client.go]: Updated

File: server/http_client.go (hunk 2 above)
Comment (bhakiyakalimuthu) [reply server/http_client.go]: Updated

File: server/request_processor.go (hunk 3 above)
Comment (bhakiyakalimuthu) [reply server/request_processor.go]: Updated

File: server/util.go (hunk 4 above)
Comment (bhakiyakalimuthu) [reply server/util.go]: Updated

File: server/http_client.go (hunk 5 above)
Comment (bhakiyakalimuthu) [reply server/http_client.go]: Updated

File: server/http_client.go (hunk 6 above)
Comment (bhakiyakalimuthu) [reply server/http_client.go]: Updated

File: server/http_client.go (hunk 7 above)
Comment (bhakiyakalimuthu) [reply server/http_client.go]: Updated

**Pull request merged**